source/common/api/posix/os_sys_calls_impl.cc

```diff
@@ -296,5 +297,61 @@ SysCallBoolResult OsSysCallsImpl::socketTcpInfo([[maybe_unused]] os_fd_t sockfd,
   return {false, EOPNOTSUPP};
 }
 
+bool OsSysCallsImpl::supportsGetifaddrs() const {
+// TODO: eliminate this branching by upstreaming an alternative Android implementation
+// e.g.: https://github.com/envoyproxy/envoy-mobile/blob/main/third_party/android/ifaddrs-android.h
+#if defined(__ANDROID_API__) && __ANDROID_API__ < 24
+  if (alternate_getifaddrs_.has_value()) {
```

Comment:
Shouldn't you be doing this in all cases otherwise the interface docs are misleading?

Reply:
@mattklein123 I think I might be misunderstanding your point. Where else should I be checking this that is not checked?

Reply:
You should check this outside of the #ifdef ?

Reply:
I see what you mean now @mattklein123. My thought was that in posix the class would default to true for _supporting_ getifaddrs regardless of if an alternate implementation is given. Only for Android<24 do we need to check for existence of an alternate implementation.

Reply:
OK I see. Makes sense. Can you just add a small comment?

/wait